### PR TITLE
feat(telemetry): add integration tests

### LINKED ISSUE #770: `kedro-telemetry`: Add Integration tests to ensure telemetry event sends only once per command
## Description

Following the PR [#766](https://github.com/kedro-org/kedro-plugins/pull/766), we now send only one Heap event per command, rather than three as previously. We have separated the integration tests to verify this behavior, which will be addressed in the current issue. A draft solution is provided in the linked PR.

### PR DESCRIPTION
**Pull request by DimedS**
## Description
Integration tests for the Kedro telemetry plugin ensure that only one event is sent to Heap Analytics per Kedro command. To facilitate this, a dummy project was created, and the tests verify the behaviour of the `run` and `info` CLI commands, as well as a session run without using the CLI.

Originally made by @noklam , was separated from https://github.com/kedro-org/kedro-plugins/pull/766, closes #770 

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro-telemetry/tests/integration/dummy-project/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# dummy_project
```
Comment (noklam) [inline kedro-telemetry/tests/integration/dummy-project/README.md:1]: can we remove this fail?

File: kedro-telemetry/tests/integration/dummy-project/stats.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"dummy": {"rows": 1, "columns": 3, "file_size": 11}}
```
Comment (noklam) [inline kedro-telemetry/tests/integration/dummy-project/stats.json:1]: why is this here?

File: kedro-telemetry/tests/integration/test_telemetry.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        mocked_heap_call = mocker.patch("kedro_telemetry.plugin._send_heap_event")
+        mocker.patch(
+            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
+        )
+        kedro_cli = KedroCLI(dummy_project_path)
+        CliRunner().invoke(kedro_cli, ["run"])
+        CliRunner().invoke(kedro_cli, ["run"])
+        CliRunner().invoke(kedro_cli, ["run"])
+        mocked_heap_call.assert_called_once()
+        raise ValueError("It should fail!")
```
Comment (noklam) [inline kedro-telemetry/tests/integration/test_telemetry.py:82]:
```suggestion
    def test_telemetry_sent_multi_with_other_kedro_command(
        self, mocker, dummy_project_path
    ):
        from kedro_telemetry.plugin import telemetry_hook

        telemetry_hook.consent = None
        telemetry_hook._sent = False
        telemetry_hook.event_properties = None
        telemetry_hook.project_path = None

        mocked_heap_call = mocker.patch("kedro_telemetry.plugin._send_heap_event")
        mocker.patch(
            "kedro_telemetry.plugin._check_for_telemetry_consent", return_value=True
        )
        kedro_cli = KedroCLI(dummy_project_path)
        CliRunner().invoke(kedro_cli, ["run"])
        CliRunner().invoke(kedro_cli, ["run"])
        CliRunner().invoke(kedro_cli, ["run"])
        mocked_heap_call.assert_called_once()
        raise ValueError("It should fail!")
```

I take the liberty to add this test directly into the PR since I think it's easier to show. This test should fail but passing for some reason

File: kedro-telemetry/tests/integration/dummy-project/README.md (hunk 1 above)
Comment (DimedS) [reply kedro-telemetry/tests/integration/dummy-project/README.md]: removed file

File: kedro-telemetry/tests/integration/dummy-project/stats.json (hunk 2 above)
Comment (DimedS) [reply kedro-telemetry/tests/integration/dummy-project/stats.json]: thanks, removed

File: kedro-telemetry/tests/integration/test_telemetry.py (hunk 3 above)
Comment (noklam) [reply kedro-telemetry/tests/integration/test_telemetry.py]: I see, the intend of having duplicate `CliRunner().invoke(kedro_cli, ["run"])` is to stimulate `kedro run` multiple times (expected multiple event).

One immediate thought came to mind, does that mean if someone is doing multiple `session.run` in a notebook, we only receive the 1st telemetry event?

Comment (DimedS) [reply kedro-telemetry/tests/integration/test_telemetry.py]: Thanks, @noklam . I tried running a session multiple times in a notebook, but I didn’t know that it’s actually prohibited (see screen). When I reloaded Kedro using the `%reload_kedro` command, the next `session.run()` did send data to telemetry, so it looks like we don’t have this issue in real usage. It seems to only be a problem in the testing environment. Therefore, I propose we remove the last test: `test_telemetry_sent_multi_with_other_kedro_command`.
Screenshot 2024-09-23 at 11 06 01

**Review by noklam — APPROVED** [review]

**Review by ElenaKhaustova — APPROVED** [review]
Thank you, @DimedS! LGTM ✨
